### PR TITLE
#3667 Add Dial Success Metric

### LINKED ISSUE #3667: Total peer additions metric
### Describe the feature

Geth has the following metric:
 * `p2p/dials/success` - the total number of dials that result in a peer being added to the peerset

It would be great to add this to reth as well!

cc @maxwolff

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by maxwolff**
adds a metric for how many outgoing dials resulted in an active session
closes [issue](https://github.com/paradigmxyz/reth/issues/3667)

### CODE REVIEW

File: crates/net/network/src/session/mod.rs [hunk 1]
```diff
@@ -473,6 +477,11 @@ impl SessionManager {
                 self.active_sessions.insert(peer_id, handle);
                 self.counter.inc_active(&direction);
 
+                match direction {
```
Comment (mattsse) [inline crates/net/network/src/session/mod.rs:480]: lets also add `is_outgoing` here

https://github.com/paradigmxyz/reth/blob/6799fc3600601d4aadd73ea808734d8bd7f11cda/crates/net/network/src/session/mod.rs#L694-L697

and use this

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive, 

style nit

File: crates/net/network/src/session/mod.rs (hunk 1 above)
Comment (maxwolff) [reply crates/net/network/src/session/mod.rs]: cool done

**Review by mattsse — APPROVED** [review]
lgtm

can also add this to the dashboard

**Pull request merged**